quickwit/quickwit-integration-tests/src/test_utils/cluster_sandbox.rs

```diff
@@ -187,6 +188,68 @@ impl ClusterSandbox {
         Ok(())
     }
 
+    // Waits for the needed number of indexing pipeline to start.
+    pub async fn wait_for_indexing_pipelines(
+        &self,
```

Comment:
I opened a ticket to deal with those...
https://github.com/quickwit-oss/quickwit/issues/3214

The original is pretty terrible by the way...
Why a while loop ? why the condition at the end?
Why the condition at the beginning?